### PR TITLE
adds a get course by code endpoint

### PR DESCRIPTION
**Pull request by Opio-Emmanuel-Omona**
##### What does this PR do?
- [x] adds a get course endpoint
- [x] returns nil when no course corresponding to the code is passed
- [x] returns a course JSON object when found

##### Note:
also added a database cleaner for the tests

### CODE REVIEW

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -17,6 +17,11 @@ def create
     end
   end
 
+  def show
+    course = Course.find_by(courseCode: params['id'])
+    render json: {course: course}
+  end
+
```
Comment (czarjulius) [inline app/controllers/courses_controller.rb]: This is a good implementation but I suggest we work with id instead of course code.

Comment (Opio-Emmanuel-Omona) [reply app/controllers/courses_controller.rb]: @czarjulius thanks for that.
I have changed it to ID

**Review by czarjulius — COMMENTED** [review]
Nice

File: Gemfile [hunk 2]
```diff
@@ -33,6 +33,7 @@ group :development, :test do
   gem 'rspec-rails'
   gem 'faker'
   gem 'factory_bot_rails'
+  gem 'database_cleaner'
```
Comment (kenneth051) [inline Gemfile:36]: why database cleaner?

**Review by czarjulius — APPROVED** [review]

File: app/controllers/courses_controller.rb [hunk 3]
```diff
@@ -17,6 +17,11 @@ def create
     end
   end
 
+  def show
+    course = Course.find_by(id: params['id'])
```
Comment (kenneth051) [inline app/controllers/courses_controller.rb]: why don't you use find_by_id?

File: spec/factories.rb [hunk 4]
```diff
@@ -6,4 +6,10 @@
         password { Faker::Crypto.md5 }
         user_type { Faker::Name.name }
     end
+
+    factory :course, class: Course do
+        courseCode { Faker::Code.asin }
+        name { Faker::Lorem.word }
+        description { Faker::Lorem.sentence }
+    end
 end
```
Comment (kenneth051) [inline spec/factories.rb]: add a new line here

File: spec/requests/courses_request_spec.rb [hunk 5]
```diff
@@ -46,4 +46,29 @@
...
+
+    it "returns the course " do
+      get "/courses/#{course_id}", headers: headers
+      courses = JSON.parse(response.body)
+      expect(courses["course"]["courseCode"]).to eq(course_code)
+      expect(courses["course"]["name"]).to eq(course_name)
+      expect(courses["course"]["description"]).to eq(course_description)
+    end
+  end
 end
```
Comment (kenneth051) [inline spec/requests/courses_request_spec.rb]: add a new line here too

File: Gemfile (hunk 2 above)
Comment (Opio-Emmanuel-Omona) [reply Gemfile:36]: I'm using database cleaner to clean the database after every test

**Review by naush — APPROVED** [review]
LGTM

**Pull request merged**